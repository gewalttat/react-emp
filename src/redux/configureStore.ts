/* eslint-disable @typescript-eslint/ban-ts-comment */
import rootReducer, { rootSaga } from './rootReducer';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware, { END } from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const createStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
  });

  // @ts-ignore
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  // @ts-ignore
  store.close = () => store.dispatch(END);

  return store;
};

export default createStore;