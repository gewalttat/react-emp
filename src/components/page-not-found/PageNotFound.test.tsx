import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import {PageNotFound} from './PageNotFound';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

describe('With React Testing Library', () => {
  it('input renders correctly', () => {
    const tree = renderer.create(<PageNotFound/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});