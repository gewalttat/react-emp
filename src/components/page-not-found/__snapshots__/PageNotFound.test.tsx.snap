// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library input renders correctly 1`] = `
Array [
  <header
    className="top-header"
  />,
  <div>
    <div
      className="starsec"
    />
    <div
      className="starthird"
    />
    <div
      className="starfourth"
    />
    <div
      className="starfifth"
    />
  </div>,
  <div
    className="lamp__wrap"
  >
    <div
      className="lamp"
    >
      <div
        className="cable"
      />
      <div
        className="cover"
      />
      <div
        className="in-cover"
      >
        <div
          className="bulb"
        />
      </div>
      <div
        className="light"
      />
    </div>
  </div>,
  <section
    className="error"
  >
    <div
      className="error__content"
    >
      <div
        className="error__message message"
      >
        <h1
          className="message__title"
        >
          Page Not Found
        </h1>
        <p
          className="message__text"
        >
          We sorry, the page you were looking for isnt found here. The link you followed may either be broken or no longer exists. Please try again, or take a look at our.
        </p>
      </div>
      <div
        className="error__nav e-nav"
      >
        <a
          className="e-nav__link"
          href="/search"
          target="_self"
        />
      </div>
    </div>
  </section>,
]
`;
